test/e2e/performance_basic_test.go

```diff
@@ -0,0 +1,204 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/kong/kubernetes-ingress-controller/v3/test"
+	"github.com/kong/kubernetes-ingress-controller/v3/test/internal/helpers"
+)
+
+var (
+	defaultResNum = 10000
+	resNumStr     = os.Getenv("PERF_RES_NUMBER")
+)
+
+// -----------------------------------------------------------------------------
+// E2E Performance tests
+// -----------------------------------------------------------------------------
+
+// TestBasicHTTPRoute will create a basic HTTP route and test its functionality
+// against a Kong proxy. This test will be used to measure the performance of
+// the KIC with OpenTelemetry.
+func TestBasicPerf(t *testing.T) {
+	t.Log("configuring all-in-one-dbless.yaml manifest test")
+	t.Parallel()
+	ctx, env := setupE2ETest(t)
+
+	if resNumStr != "" {
+		if num, err := strconv.Atoi(resNumStr); err == nil {
+			defaultResNum = num
+		}
+	}
+
+	t.Log("deploying kong components")
+	ManifestDeploy{Path: dblessPath}.Run(ctx, t, env)
+
+	t.Log("deploying a minimal HTTP container deployment to test Ingress routes")
+	container := generators.NewContainer("httpbin", test.HTTPBinImage, test.HTTPBinPort)
+	deployment := generators.NewDeploymentForContainer(container)
+	deployment, err := env.Cluster().Client().AppsV1().Deployments("default").Create(ctx, deployment, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	t.Logf("exposing deployment %s via service", deployment.Name)
+	service := generators.NewServiceForDeployment(deployment, corev1.ServiceTypeLoadBalancer)
+	_, err = env.Cluster().Client().CoreV1().Services("default").Create(ctx, service, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	kubeconfig := getTemporaryKubeconfig(t, env)
+
+	ingressTpl := `
+---
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: test-ingress-%d
+spec:
+  ingressClassName: kong
+  rules:
+  - host: example-%d.com
+    http:
+      paths:
+      - backend:
+          service:
+            name: httpbin
+            port:
+              number: 80
+        path: /get
+        pathType: Exact
+
+`
+
+	ingressYaml := ""
+	for i := 0; i < defaultResNum; i++ {
+		ingressYaml += fmt.Sprintf(ingressTpl, i, i)
+	}
+
+	startTime := time.Now()
+	err = applyResourceWithKubectl(ctx, t, kubeconfig, ingressYaml)
+	require.NoError(t, err)
+	completionTime := time.Now()
+
+	t.Log("getting kong proxy IP after LB provisioning")
+	proxyURLForDefaultIngress := "http://" + getKongProxyIP(ctx, t, env)
+
+	t.Log("waiting for routes from Ingress to be operational")
+
+	// create wait group to wait for all ingress rules to take effect
+	randomList := getRandomList(defaultResNum)
+	var wg sync.WaitGroup
+	wg.Add(len(randomList))
+
+	for _, i := range randomList {
+		go func(i int) {
+			defer wg.Done()
+
+			require.Eventually(t, func() bool {
+				return isRouteActive(ctx, t, helpers.DefaultHTTPClient(), proxyURLForDefaultIngress, "get", fmt.Sprintf("example-%d.com", i))
+			}, ingressWait, time.Millisecond*500)
```

Comment:
I think defining the interval also as a variable is better.

Reply:
Of course, it is very simple to define the polling interval time, but I think it may not be very effective.

The polling interval will directly affect the test results. 

* If a relatively large polling interval is defined, the overall time consumption will increase, which is obviously unreasonable. 
* If a small interval is defined, it will cause a large number of concurrent requests to Kong Gateway, which may not be what we expected.

I have created 10 clients and they are intervals every 500ms. Currently, this value seems acceptable.